src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -174,14 +171,14 @@ public void RemoveAt(int index)
         {
             get
             {
-                if (index >= _size)
+                if (index < 0 || index >= _size)
```

Comment:
You can improve the performance here by writing `if ((uint)index >= (uint)_size)` instead.